src/redux/mission/mission.js

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable no-case-declarations */
+import axios from 'axios';
+
+const initialState = [];
+
+const FETCH_MISSION_SUCCESS = 'space-travelers/mission/FETCH_MISSION_SUCCESS';
+const FETCH_MISSION_FAILURE = 'space-travelers/mission/FETCH_MISSION_FAILURE';
+const RESERVE_MISSION = 'space-travelers/mission/RESERVE_MISSION';
+const CANCEL_MISSION = 'space-travelers/mission/CANCEL_MISSION';
+const URL = 'https://api.spacexdata.com/v3/missions';
+
+export const reserveMission = (payload) => ({
+  type: RESERVE_MISSION,
+  payload,
+});
+
+export const cancelMission = (payload) => ({
+  type: CANCEL_MISSION,
+  payload,
+});
+
+export const fetchMissionSuccess = (payload) => ({
+  type: FETCH_MISSION_SUCCESS,
+  payload,
+});
+
+export const fetchMissionFailure = (error) => ({
+  type: FETCH_MISSION_FAILURE,
+  payload: error,
+});
+
+export const fetchMission = () => (dispatch) => {
+  axios.get(URL).then((res) => {
+    const mission = res.data;
+    dispatch(fetchMissionSuccess(mission));
+  }).catch((err) => {
+    dispatch(fetchMissionFailure(err.message));
+  });
+};
+
+const missionReducer = (state = initialState, { type, payload }) => {
+  switch (type) {
+    case FETCH_MISSION_SUCCESS:
+      return payload;
+    case FETCH_MISSION_FAILURE:
+      return state;
+    case RESERVE_MISSION:
+      const newState = state.map((mission) => {
+        if (mission.mission_id !== payload) {
+          return mission;
+        }
+        return { ...mission, reserved: true };
+      });
+      return newState;
+    case CANCEL_MISSION:
+      const leaveState = state.map((mission) => {
+        if (mission.mission_id !== payload) {
+          return mission;
+        }
+        return { ...mission, reserved: false };
+      });
+      return leaveState;
+    default:
+      return state;
+  }
+};
+
+export default missionReducer;
```

Comment:
O[tional suggestions

- [ ] Even though the implementation is working perfectly, a bunch of code is still in one file. It's not helpful at all. Let us split the code into separate files so that we can make the component smaller.

      -  Let us create three folders under the redux folder
      
- Actions folder will keep the `action creators and action types`   

Ex of action creators: `export const allowLeaveMission = (payload) => ({
  type: LEAVE_MISSION,
  payload,
});   `

Example of action type: `const JOIN_MISSION = 'space/missions/JOIN_MISSION';`

- Reducers folder will keep the cases 

Example:

`const missionsReducer = (state = { missions: [] }, action) => {
  switch (action.type) {
    case MISSION_START:
      return {
        ...state,
        loading: true,
      };
    case MISSION_FAILED:
      return {
        ...state,
        loading: false,
        error: action.payload,
      };`
      
- thunk folder will keep the API implementations, fetch some information from the API link.
 
Example

`export const loadMissionsData = () => (dispatch) => {
  dispatch(loadingStart());
  fetchData()
    .then((res) => res.json())
    .then((result) => dispatch(loadMission(result)))
    .catch((err) => dispatch(manageFailure(err.message)));
};`

Please, let us try to split it out into different files to keep our code clean and easy to maintain.